src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/AnyOS/ManagedPal.KeyTrans.cs

```diff
@@ -223,5 +222,48 @@ internal byte[] DecryptCek(X509Certificate2 cert, RSA privateKey, out Exception
             }
 #endif
         }
+
+        private static bool TryGetRsaOaepEncryptionPadding(
+            ReadOnlyMemory<byte>? parameters,
+            out RSAEncryptionPadding rsaEncryptionPadding,
+            out Exception exception)
+        {
+            exception = null;
+            rsaEncryptionPadding = null;
+            if (parameters == null || parameters.Value.IsEmpty)
+            {
+                exception = new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
+                return false;
+            }
+            try
+            {
+                OaepParamsAsn oaepParameters = OaepParamsAsn.Decode(parameters.Value, AsnEncodingRules.DER);
+                switch (oaepParameters.HashFunc.Algorithm.Value)
+                {
+                    case Oids.Sha1:
+                        rsaEncryptionPadding = RSAEncryptionPadding.OaepSHA1;
+                        return true;
+                    case Oids.Sha256:
+                        rsaEncryptionPadding = RSAEncryptionPadding.OaepSHA256;
+                        return true;
+                    case Oids.Sha384:
+                        rsaEncryptionPadding = RSAEncryptionPadding.OaepSHA384;
+                        return true;
+                    case Oids.Sha512:
+                        rsaEncryptionPadding = RSAEncryptionPadding.OaepSHA512;
+                        return true;
+                    default:
+                        exception = new CryptographicException(
+                            SR.Cryptography_Cms_UnknownAlgorithm,
+                            oaepParameters.HashFunc.Algorithm.Value);
+                        return false;
+                }
+            }
```

Comment:
MGF and PSourceFunc need to be verified as understood-to-be-the-default.

Reply:
To be clear, you just want to check the OIDs for the AlgorithmIdentifers for mgf and pSourceFunc are id-mgf1 and id-pSpecified, respectively?

Reply:
I believe that our OAEP is that the MGF hash function is the same as the message hash function, so it needs to check that it's id-mgf1 and that the digest is correct.

Likewise, it needs to be id-pSpecified, and not have a semantic value for `p`.

Essentially, we're about to process the parameters block.  Every byte in it needs to be understood to match what we're doing.